app/dao/notifications_dao.py

```diff
@@ -70,6 +71,37 @@ def dao_get_last_template_usage(template_id, template_type, service_id):
     ).first()
 
 
+@statsd(namespace="dao")
+def dao_get_last_date_template_was_used(template_id, template_type, service_id):
+    last_date = db.session.query(
+        functions.max(FactNotificationStatus.bst_date)
+    ).filter(
+        FactNotificationStatus.template_id == template_id,
+        FactNotificationStatus.key_type != KEY_TYPE_TEST
+    ).scalar()
+
+    last_date_from_notifications = db.session.query(
+        functions.max(Notification.created_at)
+    ).filter(
+        Notification.service_id == service_id,
+        Notification.notification_type == template_type,
+        Notification.template_id == template_id,
+        Notification.key_type != KEY_TYPE_TEST
+    ).scalar()
+
+    if last_date and last_date_from_notifications:
+        if datetime.combine(last_date, datetime.utcnow().min.time()) >= last_date_from_notifications:
+            return last_date
```

Comment:
is this ever going to be the case?

> "we have more recent ft_notification_status data than what is in the `Notification` table"

i don't think we need this check at all, and should always just return last_date_from_notifications in this case.

then all of this can just be `return last_date_from_notifications or last_date`